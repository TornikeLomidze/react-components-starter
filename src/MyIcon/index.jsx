import React from 'react'
import PropTypes from 'prop-types'
import { ReactComponent as Attachment }  from './icons/attachment.svg'

const MyIcon = props => {

  switch (props.source) {
    case 'attachment':
      return <Attachment />
    default:
      return <span />
  }
}

MyIcon.defaultProps = {}

export const IconPropTypes = {
  source: PropTypes.string.isRequired,
}

MyIcon.propTypes = IconPropTypes

export default MyIcon
