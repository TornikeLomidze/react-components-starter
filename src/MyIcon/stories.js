import React from 'react'
import { storiesOf } from '@storybook/react'
import VHIcon from '.'

storiesOf('Source|Icon/Attachment', module)

.add('Icon', () => (
    <VHIcon source="attachment" />
))

