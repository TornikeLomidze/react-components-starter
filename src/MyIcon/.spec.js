import React from 'react'
import MyIcon from '.';
import { render } from '../../test.setup';
import renderer from 'react-test-renderer';


describe('MyIcon', () => {
  it('renders component without crashing', () => {
    const tree = renderer.create(<MyIcon source='test' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});