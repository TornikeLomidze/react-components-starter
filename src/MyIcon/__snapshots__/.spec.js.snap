// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyIcon renders component without crashing 1`] = `<span />`;
